db/memtable_list.cc

```diff
@@ -281,12 +514,16 @@ bool MemTableList::IsFlushPending() const {
 }
 
 // Returns the memtables that need to be flushed.
-void MemTableList::PickMemtablesToFlush(autovector<MemTable*>* ret) {
+void MemTableList::PickMemtablesToFlush(const uint64_t* max_memtable_id,
+                                        autovector<MemTable*>* ret) {
   AutoThreadOperationStageUpdater stage_updater(
       ThreadStatus::STAGE_PICK_MEMTABLES_TO_FLUSH);
   const auto& memlist = current_->memlist_;
   for (auto it = memlist.rbegin(); it != memlist.rend(); ++it) {
     MemTable* m = *it;
+    if (max_memtable_id != nullptr && m->GetID() > *max_memtable_id) {
+      continue;
```

Comment:
Shouldn't they be ordered by ID such that we only pick a contiguous range including the last memtable?

Reply:
Yes the memtables in the imm list are ordered by ID.
The argument `max_memtable_id` aims to pick memtables with IDs smaller than this value. Therefore, I think we still need to traverse the list.
It's possible that the current flush job should NOT pick the current, last memtable in the list.

Reply:
We discussed offline, and have confirmed the replacement of `continue` with `break`.